sources/packages/backend/libs/integrations/src/services/student/student.service.ts

```diff
@@ -122,21 +122,24 @@ export class StudentService extends RecordDataModelService<Student> {
    * @param sinValidation SIN validation record to have the
    * relationship created with the student.
    * @param auditUserId user that should be considered the one that is causing the changes.
+   * @param transactionalEntityManager entity manager to execute in transaction.
    * @returns updated student.
    */
   async updateSINValidationByStudentId(
     studentId: number,
     sinValidation: SINValidation,
     auditUserId: number,
+    transactionalEntityManager: EntityManager,
   ): Promise<Student> {
-    const studentToUpdate = await this.repo
+    const studentToUpdate = await transactionalEntityManager
+      .getRepository(Student)
       .createQueryBuilder("student")
       .select("student.id")
       .where("student.id = :studentId", { studentId })
       .getOne();
     studentToUpdate.modifier = { id: auditUserId } as User;
     studentToUpdate.sinValidation = sinValidation;
-    return this.repo.save(studentToUpdate);
+    return transactionalEntityManager.save(studentToUpdate);
```

Comment:
Usually, we get the specific repo, and then we do the save: `transactionalEntityManager.getRepository(Student)`.
In this method, it needs to get the repo twice so the repo can be added to a const and used on both places.